doc/html/compile.xhtml

```diff
@@ -15,67 +17,190 @@
-  build the tester, and it is not linked to the library.) Change
-  directory to sleef-3.X directory and run make. The built headers
-  and libraries will be located under include and lib directories.
+1. Make sure cmake is available on the command-line.
+</p>
+<pre>$ cmake --version
+(should display a version number greater than or equal to 3.4.3)</pre>
+
+<p class="noindent">
+2. Download the tar from
+the <a class="underlined" href="https://sourceforge.net/projects/sleef/files/">software
+repository</a> or checkout out the source code from
+the <a class="underlined" href="https://github.com/shibatch/sleef">GitHub repository</a>.
+</p>
+<pre>$ git clone https://github.com/shibatch/sleef</pre>
+
+<p class="noindent">
+3. Make a separate directory to create an out-of-source build. SLEEF does not
+allow for in-tree builds.
+</p>
+<pre>$ cd sleef-project
+$ mkdir my-sleef-build &amp;&amp; cd my-sleef-build</pre>
+
+<p class="noindent">
+4. Run cmake to configure your project and generate the system to build it:
+</p>
+<pre>$ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+	-DCMAKE_INSTALL_PREFIX=../my-sleef-install \
+	..</pre>
+
+<p class="noindent">
+This flag configures an optimised libsleef shared library build with
+basic debug info.  By default, cmake will autodetect your system
+platform and configure the build using the default parameters. You can
+control and modify these parameters by setting variables when running
+cmake. See the list of <a class="underlined"
+href="#build-customization">options and variables</a> for customizing
+your build.
 </p>
 
 <p>
-  You can run make install using sudo command to install the library
-  and header. Those files are installed under /usr/lib and
-  /usr/include. You can run make uninstall to uninstall those files.
+NOTE: On **Windows**, you need to use a specific generator like this:
+`cmake -G"Visual Studio 14 2015 Win64" ..` specifying the Visual
+Studio version and targeting specifically `Win64` (to support
+compilation of AVX/AVX2) Check `cmake -G` to get a full list of
+supported Visual Studio project generators.  This generator will
+create a proper solution `SLEEF.sln` under the build directory.  You
+can still use `cmake --build .` to build without opening Visual
+Studio.
 </p>
 
-<pre class="code">
-<code>$ sudo apt-get install libmpfr-dev libgomp1-dev gcc</code>
-<code>$ cd sleef-3.X</code>
-<code>$ make</code>
-<code>$ sudo make install</code>
-</pre>
-<p style="text-align:center;">
-  Fig. 2.1: Commands for compiling SLEEF
+<p class="noindent" style="margin-top: 1em;">
+5. Now that you have the build files created by cmake, proceed from the top
+of the build directory:
 </p>
+<pre>$ make sleef</pre>
 
+<p class="noindent">
+6. Install the library under ../my-sleef/install by running:
+</p>
+<pre>$ make install</pre>
 
-<h3>Compiling library with Microsoft Visual C++</h3>
+<p class="noindent">
+7. You can execute the tests by running:
+</p>
+<pre>$ make test-libm</pre>
```

Comment:
`test-libm` does not exist as a target in the cmake configuration. The target to run the tests is `make test`, but it executes and complete correctly only if all the targets are build with `make all` or just `make`.

Is this a leftover of the old docs related to the Makefile builds?